source/deployment/ha.rst

```diff
@@ -0,0 +1,55 @@
+High Availability 
+----
+
+*Available in Enterprise Edition E20 version 3.1 and higher.*
+
+High Availability support enables a Mattermost server to maintain service during outages and hardware failures of underlying services and hardware through the use of redundant infrastructure. 
+
+- Support for high availability database configurations is available in Mattermost Enterprise Edition E20 version 3.1 and higher. 
+- Support for high availability configurations of the Mattermost application server is planned for release in Mattermost Enterprise Edition E20 during summer 2016. 
+
+Setting up an active-passive redundant database configuration
+====
+
+The Mattermost server can set up to use one "master" database and up to 8 read replica databases. Mattermost distributes read requests across all databases, and sends write requests to the master database, and those changes are then sent to update the read replicas. 
+
+Sizing databases
+^^^^
+
+Please see `documentation on sizing database servers <http://docs.mattermost.com/install/requirements.html#hardware-requirements>`_ for guidance to determine appropriate hardware. 
+
+It's highly recommended that at least one read replica have a minimum of **three times the disk space** of the master database. This way, if the master database signals a "low disk space" warning the administrator can switch over to the read replica with a larger disk and continue operation, while also having room to locally duplicate and manipulate the production database. 
+
+
+Deploying a multi-database configuration 
+^^^^
+
+To configure a multi-database Mattermost server edit the `DataSource` setting `config.json` with a list of comma-separated database connection strings, with the first entry being the connection to the master database.
```

Comment:
The master setting is a separate setting.  The read replica settings is a json string array in the form ["readreplia1", "readreplia2"]